src/cmd/ksh93/tests/builtins.sh

```diff
@@ -629,6 +629,23 @@ then
     rmdir "$pwd/f1"
 fi
 
+OLDPWDSYMLINK="$TEST_DIR/oldpwdsymlink"
```

Comment:
If ksh had a `realpath` command I would recommend using it to make this test work even if `$TEST_DIR` contains symlinks (like it does on macOS).  Instead the simplest "fix" is to probably do `cd $(pwd -P)` here and then use `$PWD` where you have `$TEST_DIR`.